File: Typescript/FacadePattern.js
// Complex subsystem classes
var CPU = /** @class */ (function () {
    function CPU() {
    }
    CPU.prototype.powerOn = function () {
        return "CPU is powered ON";
    };
    CPU.prototype.jump = function (position) {
        return "CPU is jumping to position ".concat(position);
    };
    CPU.prototype.execute = function (data) {
        return "CPU is executing '".concat(data, "'");
    };
    return CPU;
}());
var BIOS = /** @class */ (function () {
    function BIOS() {
    }
    BIOS.prototype.load = function (position, data) {
        return "BIOS loaded '".concat(data, "' into memory at position ").concat(position);
    };
    return BIOS;
}());
var HardDrive = /** @class */ (function () {
    function HardDrive() {
    }
    HardDrive.prototype.read = function (sector, size) {
        return "HardDrive read ".concat(size, " bytes from sector ").concat(sector);
    };
    return HardDrive;
}());
//Facade class
var ComputerFacade = /** @class */ (function () {
    function ComputerFacade() {
        this.cpu = new CPU();
        this.bios = new BIOS();
        this.hardDrive = new HardDrive();
    }
    ComputerFacade.prototype.start = function () {
        var result = [];
        result.push(this.cpu.powerOn());
        result.push(this.bios.load(0, "BOOT data"));
        result.push(this.cpu.jump(0));
        result.push(this.cpu.execute("BOOT data"));
        result.push(this.hardDrive.read(0, 1024));
        result.push(this.bios.load(1, "OS data"));
        result.push(this.cpu.jump(1));
        result.push(this.cpu.execute("OS data"));
        return result;
    };
    return ComputerFacade;
}());
// Client code
function main() {
    var computer = new ComputerFacade();
    var result = computer.start();
    for (var _i = 0, result_1 = result; _i < result_1.length; _i++) {
        var action = result_1[_i];
        console.log(action);
    }
}
main();

File: Python/FacadePattern.py
# Complex subsystem classes
class CPU:
    def powerOn(self):
        return "CPU is powered On"

    def jump(self, position):
        return f"CPU is jumping to position {position}"

    def execute(self,data):
        return f"CPU is executing '{data}'"


class BIOS:
    def load(self, position, data):
        return f"BIOS loaded '{data}' into memory at position {position}"


class HardDrive:
    def read(self, sector, size):
        return f"HardDrive read {size} bytes from sector {sector}"


# Facade class
class ComputerFacade:
    def __init__(self):
        self.cpu = CPU()
        self.bios = BIOS()
        self.hard_drive = HardDrive()

    def start(self):
        result = []
        result.append(self.cpu.powerOn())
        result.append(self.bios.load(0, "BOOT data"))
        result.append(self.cpu.jump(0))
        result.append(self.cpu.execute("BOOT data"))
        result.append(self.hard_drive.read(0, 1024))
        result.append(self.bios.load(1, "OS data"))
        result.append(self.cpu.jump(1))
        result.append(self.cpu.execute("OS data"))
        return result


# Client code
def main():
    computer = ComputerFacade()
    result = computer.start()
    for action in result:
        print(action)


if __name__ == "__main__":
    main()
